sdk/core/azure-core/test/ut/azure_libcurl_core_main_test.cpp

```diff
@@ -53,11 +53,21 @@ namespace Azure { namespace Core { namespace Test {
       EXPECT_TRUE(session->m_keepAlive);
       EXPECT_FALSE(session->m_connectionUpgraded);
     }
+    // here the session is destroyed and the connection is moved to the pool
+    // the same destructor also makes a call to start the cleanup thread
+    // which will sleep for DefaultCleanerIntervalMilliseconds then loop through the connections
+    // in the pool and check for the ones that are expired(DefaultConnectionExpiredMilliseconds) and remove them
+    // which will be the case here if we wait long enough. 
+    // the test is flaky due to the fact that tests in the CI pipeline might take longer than 90 sec to execute 
```

Comment:
I might rephrase this as "without a delay, the test will be flaky" or something similar.

Also, I'm not sure that the new version tests the same thing as the old test did. The old test verifies that the connection is added to the cache after the session is destroyed, the new test verifies that the cache is cleared.

I think both checks are valid and important.